### DISCUSSION TITLE
Allow entering train number directly

### DISCUSSION THREAD

**Discussion by rugk** [Traewelling/traewelling#2014] [Ideas / Feature Requests]
**Is your feature request related to a problem? Please describe.**
It is way harder for me to look up and find the nearest/last train stop, then look at the big sign or iceportal.de or so top find out my three/four-digit train number.
Also, these are way more characters to write (and make typos). And if you have written these letters completely (and correctly) I may already be at the next train stop… :stuck_out_tongue_winking_eye: 

**Describe the solution you'd like**
In the search at "Where are you?" it should allow me to enter "ICE 1234" etc.

**Describe alternatives you've considered**
Current state: I need to enter the station **and** then *additionally* select the train.
If you need the station, do it in the reverse way (optionally): If I enter/select a train first, then offer me to select the station afterwards.

Also, I see much more trains at one station usually (if you have a big main station or so), so it is really hard to select the correct one.
Entering the train number seems easy though. Then you can easily select the train stop (there likely are not that much).

**Additional context**
Of course, you can just implement this in addition to the current mechanism.

**Comment by HerrLevin** [comment]
Hi, thank you very much for your contribution.

Sadly it's technically not possible to easily fetch the train via the linenumber. At least not with our used [API](https://github.com/derhuerst/db-rest). (Even the official App "DB Navigator" did not implement this feature natively. It's just a webview for [mobile.bahn.de](https://mobile.bahn.de/bin/mobil/trainsearch.exe/dox?webview=&) )

In the future we want to implement a "location based"-feature, wherein your current station will be fetched via the geolocation of your device.
At this point of development you're able to directly access your latest visited train stations via the ![Screen Capture_select-area_20200928162305](https://user-images.githubusercontent.com/1267894/94444814-03367980-01a7-11eb-813f-553212f20dab.png) icon next to the search bar.

I hope this solution is acceptable for now. Sorry. :disappointed:

**Comment by rugk** [comment]
Well… yeah, thanks for the explanation. Usability/technically said it's not "acceptable", i.e. I'd want a different solution, but if you cannot implement it, you cannot implement it.

BTW, tagging with "good first issue" is misleading IMHO. This tag is usually used to indicate a "low requirements" bug that could be easily fixed by new contributors. See https://yourfirstpr.github.io/ and https://up-for-grabs.net/.
Maybe you'd rather want to use "help wanted".

**Comment by pascalwittler** [comment]
It seems like https://bahn.expert found a way to find trains by their train number and then list all stops of the journey, see

https://bahn.expert/details/IC%202299/2024-06-03

You can start a search by clicking at the burger menu. Then select "Zugsuche". When you enter a search term, bahn.expert sends a request to its own API:

https://bahn.expert/api/journeys/v1/find/number/2299?initialDepartureDate=2024-06-03T19:46:38.567Z&filtered=true&limit=10

There is also a search form for train numbers at https://mobile.bahn.de in the section "Ist mein Zug pünktlich?", see

https://mobile.bahn.de/bin/mobil/trainsearch.exe/dox
